### PR TITLE
Working DrawOnFreePanelFeature

### PR DESCRIPTION
**Pull request by WojFar**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+public class DrawOnFreePanelFeature {
+
+	private static Application app;
+	private static DriverManager driverMan;
+
+	public static void setupButtonClick(Application application, DriverManager driverManager) {
+		app = application;
+		driverMan = driverManager;
+
+		app.getFreePanel().addMouseListener(new MouseInputAdapter() {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java]: Extract MouseInputAdapter to separate class.

File: src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+			
+			public void mouseClicked(MouseEvent e) {
+				Point clickPoint = e.getPoint();
+				clickPoint.x -= app.getFreePanel().getSize().width/2;
+				clickPoint.y -= app.getFreePanel().getSize().height/2;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java]: app.getFreePanel() - unnecessary code duplication

**Review by kowallus — CHANGES_REQUESTED** [review]
Please apply a small refactor.

File: src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import edu.kis.powp.jobs2d.drivers.DriverManager;
+
+public class DrawOnFreePanelFeature {
+
+	private static Application app;
+
+	public static void setupButtonClick(Application application, DriverManager driverManager) {
+		app = application;
+
+		app.getFreePanel().addMouseListener(new MouseInputAdapterForSetupButtonClick(app.getFreePanel(),driverManager.getCurrentDriver()));
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java]: app.getFreePanel()....(app.getFreePanel()) it looks to me like some recursive dependency and it is difficult for programmer to come up with. I suggest introduce something like static method :
MouseInputAdapterForSetupButtonClick.enable(app.getFreePanel(), driverManager) that would encapsulate this a bit awkward code.

File: src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package edu.kis.powp.jobs2d.features;
+
+import edu.kis.powp.appbase.Application;
+import edu.kis.powp.jobs2d.drivers.DriverManager;
+
+public class DrawOnFreePanelFeature {
+
+	private static Application app;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java]: Is it really necessary here, to store Application? 
Moreover, it seems that dependency with the Application isn't really necessary.

**Review by kowallus — CHANGES_REQUESTED** [review]
Still some things are not right to me.

File: src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java [hunk 5]
```diff
@@ -5,11 +5,9 @@
 
 public class DrawOnFreePanelFeature {
 
-	private static Application app;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java]: You have forgotten to remove Application from imports (line 3)

**Review by kowallus — CHANGES_REQUESTED** [review]
OK, but fix import.

File: src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java (hunk 5 above)
Comment (WojFar) [reply src/main/java/edu/kis/powp/jobs2d/features/DrawOnFreePanelFeature.java]: It's is used in setupButtonClick() function as parameter so it is needed

**Pull request merged**